tests/src/Kernel/CollaboraMediaAccessTest.php

```diff
@@ -0,0 +1,440 @@
+            $media_entities,
+            [
+                'preview in collabora',
+                'edit in collabora',
+            ],
+        );
+    }
+
+    /**
+     * Asserts which users can access which entity operations.
+     *
+     * @param array<string, array<string, list<string>>> $expected
+     *   Expected outcome.
+     *   The array is keyed by entity key and operation.
+     *   The values are lists of keys from the $accounts parameter.
+     * @param array<string, \Drupal\Core\Session\AccountInterface> $accounts
+     *   Entities to check.
+     * @param array<string, \Drupal\Core\Entity\EntityInterface> $entities
+     *   Operations to check.
+     * @param list<string> $operations
+     *   User accounts to check.
+     * @param bool $negate
+     *   FALSE, if $expected describes access being granted.
+     *   TRUE, if $expecte describes access being denied.
+     */
+    protected function assertEntityAccess(array $expected, array $accounts, array $entities, array $operations, bool $negate = FALSE): void {
+        $actual = [];
+        foreach ($accounts as $account_key => $account) {
+            $actual[$account_key] = [];
+            foreach ($entities as $entity_key => $entity) {
+                foreach ($operations as $operation) {
+                    $has_access = $entity->access($operation, $account);
+                    if ($has_access xor $negate) {
+                        $actual[$account_key][$entity_key][] = $operation;
+                    }
+                }
+            }
+        }
+        // Use yaml to avoid integer keys in list output.
+        $this->assertSame(
+            "\n" . Yaml::encode($expected),
+            "\n" . Yaml::encode($actual),
+            $negate
+                ? 'Users without access to given entities'
+                : 'Users with access to given entities',
+        );
+    }
+
+    /**
+     * Asserts which users have access to which entity paths.
+     *
+     * @param array<string, list<string>> $expected
+     *   Array indicating which url should be accessible by which user.
+     *   The array keys are either string keys from the $paths array.
+     *   The array values are lists of keys from the $accounts array with access
+     *   to that path.
+     * @param array<string, \Drupal\Core\Session\AccountInterface> $accounts
+     *   Accounts to test access with, keyed by a distinguishable name.
+     * @param array<string, \Drupal\Core\Entity\EntityInterface> $entities
+     *   Entities for which to build paths.
+     * @param array<string, string> $sprintf_path_patterns
+     *   Path patterns with '%s' placeholder for the entity id.
+     * @param bool $negate
+     *   FALSE, if $expected describes access being granted.
+     *   TRUE, if $expected describes access being denied.
+     */
+    protected function assertEntityPathsAccess(array $expected, array $accounts, array $entities, array $sprintf_path_patterns, bool $negate = FALSE) {
```

Comment:
Yeah, i was tempted to split out methods so that individual methods become smaller.
But I think now that in these test classes, the number of methods and the call stack size can be a bigger source of complexity than the length of an individual method.